src/state.py

```diff
@@ -129,22 +131,22 @@ def _fetch_ssh_repository_key(model: ops.Model, config: WazuhConfig) -> str | No
     return custom_config_ssh_key_content
 
 
-def _fetch_agent_password(model: ops.Model, config: WazuhConfig) -> str | None:
-    """Fetch the password for the agent.
+def _fetch_password(model: ops.Model, secret_id: str | None) -> str | None:
+    """Fetch the password for the a given secret ID.
 
     Args:
         model: the Juju model.
-        config: the charm configuration.
+        secret_id: the secret ID.
 
-    Returns: the SSH key for the repository, if any.
+    Returns: the password stored in the secret, if any.
 
     Raises:
         InvalidStateError: if the secret when the key should reside is invalid.
     """
     agent_password_content = None
-    if config.agent_password:
+    if secret_id:
```

Comment:
If secret_id is None...should be an error? or a warning?

Reply:
No, I'd say it's ok as it is. You don't provide an id, nothing happens, `None` is returned